addons/mail/wizard/mail_compose_message.py

```diff
@@ -518,21 +510,25 @@ def _process_state(self, mail_values_dict):
                 mail_values['failure_type'] = 'mail_bl'
                 # Do not post the mail into the recipient's chatter
                 mail_values['is_notification'] = False
-            # void of falsy values -> error
-            elif not mail_to:
+            # void or falsy values -> error
+            elif not any(recipients['mail_to']):
                 mail_values['state'] = 'cancel'
                 mail_values['failure_type'] = 'mail_email_missing'
-            elif not mail_to_normalized:
+            elif not any(recipients['mail_to_normalized']):
                 mail_values['state'] = 'cancel'
                 mail_values['failure_type'] = 'mail_email_invalid'
-            elif optout_emails and mail_to in optout_emails:
+            elif optout_emails and all(
+                    mail_to in optout_emails for mail_to in recipients['mail_to']
+                ):
                 mail_values['state'] = 'cancel'
                 mail_values['failure_type'] = 'mail_optout'
-            elif done_emails and mail_to in done_emails and not mailing_document_based:
+            elif done_emails and not mailing_document_based and all(
+                    mail_to in done_emails for mail_to in recipients['mail_to']
+                ):
```

Comment:
Would say that any email in blocklist win, but that's more a personal choice. Not sure if this was discussed ?

Reply:
It has not been discussed, my opinion is that customers can still create a filtrer in their own mailbox if they *really* don't want to receive those emails, but the opposite is impossible. It also is an improvement over the current system, that just send all emails when there are multiple recipients. Changing the behavior for `any` would be more disruptive, why not in master, not sure in stable.